### PR TITLE
Bump version numbers for LTS patch release

### PR DESCRIPTION
**Pull request by schoims**
| packages | updated version | 
|:--------------|:--------------:|
|Microsoft.Azure.Devices.Client| 1.36.7 |
| Microsoft.Azure.Devices |   1.31.4 |
|Microsoft.Azure.Devices.Shared | 1.27.2 |
|Microsoft.Azure.Devices.Provisioning.Transport.Mqtt | 1.14.3 |
|Microsoft.Azure.Devices.Provisioning.Service |   1.16.5 |

## Microsoft.Azure.Devices.Client 1.36.7
- Update DotNetty versions (https://github.com/Azure/azure-iot-sdk-csharp/pull/2970)
- Bring amqp sas token refresh cleanup logic to LTS (https://github.com/Azure/azure-iot-sdk-csharp/pull/3039)
- Setting MaxDepth parameter in the JsonSerializerSettings as safe defaults #3045

## Microsoft.Azure.Devices  1.31.4
- Setting MaxDepth parameter in the JsonSerializerSettings as safe defaults #3045

## Microsoft.Azure.Devices.Shared 1.27.2
- Setting MaxDepth parameter in the JsonSerializerSettings as safe defaults #3045

## Microsoft.Azure.Devices.Provisioning.Transport.Mqtt 1.14.3
- Update DotNetty versions (https://github.com/Azure/azure-iot-sdk-csharp/pull/2970)
- Add M2M message failure fix (https://github.com/Azure/azure-iot-sdk-csharp/pull/3006)

## Microsoft.Azure.Devices.Provisioning.Service 1.16.5
- Setting MaxDepth parameter in the JsonSerializerSettings as safe defaults #3045

### CODE REVIEW

**Review by drwill-ms — APPROVED** [review]

**Comment by drwill-ms** [comment]
Be sure to add a row in our 1N Release tab on Hub/DPS .NET version 1.x tracking

File: versions.csv [hunk 1]
```diff
@@ -1,10 +1,10 @@
 AssemblyPath, Version
-iothub\device\src\Microsoft.Azure.Devices.Client.csproj, 1.36.6
-iothub\service\src\Microsoft.Azure.Devices.csproj, 1.31.3
-shared\src\Microsoft.Azure.Devices.Shared.csproj, 1.27.1
-provisioning\device\src\Microsoft.Azure.Devices.Provisioning.Client.csproj, 1.16.4
+iothub\device\src\Microsoft.Azure.Devices.Client.csproj, 1.36.7
+iothub\service\src\Microsoft.Azure.Devices.csproj, 1.31.4
+shared\src\Microsoft.Azure.Devices.Shared.csproj, 1.27.2
+provisioning\device\src\Microsoft.Azure.Devices.Provisioning.Client.csproj, 1.16.5
```
Comment (abhipsaMisra) [inline versions.csv:5]: This csproj update seems to be missing.

Comment (abhipsaMisra) [reply versions.csv:5]: If this project isn't required to be updated, then you'll need to remove this entry from the .csv file

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**